masonry_core/src/widgets/text_area.rs

```diff
@@ -1018,16 +1034,33 @@ impl<const EDITABLE: bool> Widget for TextArea<EDITABLE> {
     }
 }
 
+/// When to insert a newline in a text area
+#[derive(Clone, Copy, Debug)]
+pub enum InsertNewline {
+    /// Insert a newline when the user presses Enter
+    ///
+    /// Note that if this is enabled, then the text area will never emit a `TextEntered` event
+    OnEnter,
+    /// Insert a newline when the user presses Shift+Enter
+    OnShiftEnter,
+    /// Never insert a newline
+    Never,
+}
```

Comment:
I think this is a reasonable-ish set of options. It feels like longer-term you'd want something more customisable, but just getting something landed is worth doing.